python/mage/tgn/definitions/raw_message_store.py

```diff
@@ -0,0 +1,30 @@
+from typing import List, Dict
+
+from mage.tgn.definitions.messages import RawMessage
+
+
+class RawMessageStore:
+    """
+    This class represents store for instances of Raw Messages
+    """
+
+    def __init__(
+        self, edge_raw_message_dimension: int, node_raw_message_dimension: int
+    ):
+        self.message_container: Dict[int, List[RawMessage]] = {}
+        self.edge_raw_message_dimension = edge_raw_message_dimension
+        self.node_raw_message_dimension = node_raw_message_dimension
+
+    def detach_grads(self) -> None:
+        for _, messages in self.message_container.items():
+            for message in messages:
+                message.detach_memory()
+
+    def get_messages(self) -> Dict[int, List[RawMessage]]:
+        return self.message_container
+
+    def update_messages(self, new_node_messages: Dict[int, List[RawMessage]]) -> None:
+        for node in new_node_messages:
+            if node not in self.message_container:
+                self.message_container[node] = []
```

Comment:
Can be `defaultdict` and then you wouldn't check the existence

Reply:
Nice!